classes/Form.php

```diff
@@ -699,6 +705,13 @@ public function post()
         $queue = $session->getFlashObject('files-upload');
         $queue = $queue[base64_encode($url)];
         if (is_array($queue)) {
+            // Allow plugins to implement additional / alternative logic
+            // Add post to event data
+            $event = $grav->fireEvent('onFormStoreUploads', new Event(['queue' => $queue, 'form' => $this, 'post' => $post]));
+            // only queue imports here
+            if (isset($event->queue)) {
```

Comment:
Another way to do this is to have just `new Event(['queue' => &$queue`... I have no preference on this one.

Reply:
From my viewpoint, there's no need to handle the post object in the uploadFile function and in the queue processing neither. The most important opportunity offered is to change the destination in the uploads section and to offer the possibility to clear destinations and apply naming needs within the second one or even assume the storage completely. These two events deal with uploads, not with the form and post processing. If it's about the rest of the form handling, it's probably better to fire a dedicated event for this. In fact, the validation event could be used for this already.

Regarding $post, @rhukster pointed out that it is probably enough to expose the form name to allow better identification and handling if plugins hook into the event. But who knows, people may come up with implementations we are not able to think of right now and it may help them to have the information available.

I will change the syntax to reference syntax as you pointed out to make it easier to use the events.